### PR TITLE
Added feature to set custom user defaults for Shared extensions support

### PR DESCRIPTION
**Pull request by suriksarkisyan**

### CODE REVIEW

File: Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
+  [self.customUserDefaults setBool:value forKey:key];
   [self.userDefaults setBool:value forKey:key];
 }
 
 - (BOOL)loadBoolforKey:(NSString *)key {
-  return [self.userDefaults boolForKey:key];
+  BOOL result = NO;
+  if (self.customUserDefaults) {
+    result = [self.customUserDefaults boolForKey:key];
+    if (!result) {
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m:21]: What if there was stored value and it was "false"?

File: Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m [hunk 2]
```diff
@@ -45,26 +55,46 @@ - (id)loadObjectForKey:(NSString *)key {
...
 
 - (NSString *)loadStringForKey:(NSString *)key {
-  return [self.userDefaults stringForKey:key];
+  NSString *result;
+  if (self.customUserDefaults) {
+    result = [self.customUserDefaults stringForKey:key];
+    
+    if (!result) {
+      result = [self.userDefaults stringForKey:key];
+    }
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m:69]: Don't we put the loaded value to `customUserDefaults` here, as we did in previous methods?

File: Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m [hunk 3]
```diff
@@ -45,26 +55,46 @@ - (id)loadObjectForKey:(NSString *)key {
...
 
 - (double)loadDoubleForKey:(NSString *)key {
-  return [self.userDefaults doubleForKey:key];
+  double result;
+  if (self.customUserDefaults) {
+    result = [self.customUserDefaults doubleForKey:key];
+    
+    if (result == 0.0) {
+      result = [self.userDefaults doubleForKey:key];
+    }
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m:84]: the same question here

File: Sources/Qonversion/Qonversion/Main/QNProductCenterManager/QNProductCenterManager.m [hunk 4]
```diff
@@ -440,15 +455,18 @@ - (void)executeEntitlementsBlocksWithError:(NSError *)error {
...
+        [self fireEntitlementsBlocks:[_blocks copy] result:@{} error:error];
+      } else {
+        [self fireEntitlementsBlocks:[_blocks copy] result:cachedEntitlements error:error];
       }
     } else {
       [self prepareEntitlementsResultWithCompletion:^(NSDictionary<NSString *,QONEntitlement *> * _Nonnull result, NSError * _Nullable error) {
-        for (QONEntitlementsCompletionHandler block in _blocks) {
-          run_block_on_main(block, result ?: @{}, error);
-        }
+        [self fireEntitlementsBlocks:[_blocks copy] result:cachedEntitlements error:error];
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Qonversion/Main/QNProductCenterManager/QNProductCenterManager.m:469]:
```suggestion
        [self fireEntitlementsBlocks:[_blocks copy] result:result ?: @{}, error:error];
```

File: Sources/Qonversion/Qonversion/Main/QONRemoteConfigManager/QONRemoteConfigManager.m [hunk 5]
```diff
@@ -22,7 +22,7 @@ @interface QONRemoteConfigManager ()
 
 @implementation QONRemoteConfigManager
 
-- (instancetype)init {
+- (instancetype)initWithRemoteConfigService:(QONRemoteConfigService *)remoteConfigService {
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Qonversion/Main/QONRemoteConfigManager/QONRemoteConfigManager.m:25]: Argument added but not used

File: Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m (hunk 1 above)
Comment (suriksarkisyan) [reply Sources/Qonversion/Qonversion/Core/QNUserDefaultsStorage/QNUserDefaultsStorage.m:21]: `if (!result)` is the same as `if (result == false)`.

**Review by SpertsyanKM — APPROVED** [review]

**Pull request merged**